lib/cdo/share_filtering.rb

```diff
@@ -76,7 +77,7 @@ def self.extract_text_blockly(program)
     end
 
     # Texts will include field values, text values in block inputs, comments, and variable names.
-    json = JSON.parse(stripped)
+    json = JSON.parse(stripped, max_nesting: DCDO.get('share_filtering_blockly_json_max_depth', JSON_MAX_DEPTH))
```

Comment:
If this fails, we will fail to load the project, right? This is maybe a longer-term fix but I wonder if we should still allow the project to load, but not allow it to be shared?

Reply:
I think as an intermediate step having it load but not be shareable (if that is easy to do here) seems like the right call. If we can't parse it, we don't know if it's shareable.

Reply:
I think that makes sense. I'll add a new `FailureType` for `PROGRAM_TOO_LARGE`.
Just noting that in `activites_controller`, we have a `rescue` for `WebPurify::TextTooLongError` and the program is still allowed. 
https://github.com/code-dot-org/code-dot-org/blob/a98a05523ea620b9ed4691a4f333206cf13c7d2e/dashboard/app/controllers/activities_controller.rb#L61-L71

Reply:
maybe worth checking with Sam?